lib/stat_tracker.rb

```diff
@@ -32,4 +32,68 @@ def highest_total_score
   def lowest_total_score
     @game_csv.map { |row| row[:away_goals].to_i + row[:home_goals].to_i }.min
   end
-end
\ No newline at end of file
+
+  def percentage_home_wins
+    home_wins = 0
+    total_wins = 0
+    @game_csv.each do |row|
+      home_wins += 1 if row[:home_goals].to_f > row[:away_goals].to_f
+      total_wins += 1
+    end
+    return (home_wins.to_f/total_wins.to_f).round(2)
+  end
+
+  def percentage_visitor_wins
+    visitor_wins = 0
+    total_wins = 0
+    @game_csv.each do |row|
+      visitor_wins += 1 if row[:home_goals].to_f < row[:away_goals].to_f
+      total_wins += 1
+    end
+    return (visitor_wins.to_f/total_wins.to_f).round(2)
+  end
+
+  def percentage_ties
+    ties = 0
+    total_wins = 0
+    @game_csv.each do |row|
+      ties += 1 if row[:home_goals].to_f == row[:away_goals].to_f
+      total_wins += 1
+    end
+    return (ties.to_f/total_wins.to_f).round(2)
+  end
+
+  def count_of_games_by_season
+    seasons_with_games = Hash.new(0)
+    @game_csv.each do |row|
+      seasons_with_games[row[:season]] += 1
+    end
+    seasons_with_games
+  end
+
+  def average_goals_per_game
+    total_games = 0
+    average = 0
+    total_goals = 0
+    @game_csv.each do |row|
+     total_goals += row[:home_goals].to_i + row[:away_goals].to_i
+      total_games += 1
+      average = (total_goals.to_f/total_games.to_f).round(2)
+    end
+    average
+  end
+
+  def average_goals_by_season
+    seasons_with_games = Hash.new(0)
+    @game_csv.each do |row|
+      seasons_with_games[row[:season]] += (row[:away_goals].to_f + row[:home_goals].to_f)
+    end
+    seasons_averages = Hash.new(0)
+    seasons_with_games.map do |season_id, total_games|
```

Comment:
Why do we use `map` instead of `each` here? Are we using the array for anything?